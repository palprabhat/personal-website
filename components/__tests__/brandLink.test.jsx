import { render } from "@testing-library/react";
import BrandLink from "../brandLink";
import { brandLinkTest } from "../../constants/testIds.const";

const testLink = "https://prabhatpal.com/";
const testLabel = "Test Label";

describe("BrandLink Component", () => {
  it("renders correct content", () => {
    const { baseElement, queryByTestId } = render(
      <BrandLink testId={brandLinkTest} link={testLink} ariaLabel={testLabel} />
    );

    const link = queryByTestId(brandLinkTest);

    expect(link).toBeInTheDocument();
    expect(link).toHaveAttribute("href", testLink);
    expect(link).toHaveAttribute("aria-label", testLabel);

    expect(baseElement).toMatchSnapshot();
  });
});
