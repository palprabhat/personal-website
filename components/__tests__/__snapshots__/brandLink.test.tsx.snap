// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BrandLink Component renders correct content 1`] = `
<div>
  <a
    aria-label="Test Label"
    class=" text-3xl text-primary-500 hover:text-primary-600 focus:text-primary-600 active:text-primary-700"
    data-testid="intro-description"
    href="https://prabhatpal.com/"
    rel="noreferrer"
    target="_blank"
  />
</div>
`;
